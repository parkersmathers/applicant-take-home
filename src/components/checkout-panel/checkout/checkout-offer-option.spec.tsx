import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import CheckoutOfferOption from './checkout-offer-option';

describe('Test Checkout Offer Option Component', () => {
    const onClick = jest.fn();
    const cost = 4900;
    const checkoutId = 'test-checkout-id';

    // Matches snapshot
    test('Component matches snapshot', () => {
        const { asFragment } = render(
            <CheckoutOfferOption checkoutId={checkoutId} cost={cost} clickHandler={onClick} />,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test('Component can call on click', () => {
        render(<CheckoutOfferOption checkoutId={checkoutId} cost={cost} clickHandler={onClick} />);

        expect(onClick).toHaveBeenCalledTimes(0);

        userEvent.click(screen.getByTestId('button'));

        expect(onClick).toHaveBeenCalledTimes(1);
    });
});
