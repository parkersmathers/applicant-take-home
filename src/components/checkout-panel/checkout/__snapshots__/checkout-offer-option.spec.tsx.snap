// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkout Offer Option Component Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="$49.00"
    class="button button--small mr-xs mb-xs"
    data-testid="button"
    id="test-checkout-id"
    title="$49.00"
    type="button"
  >
    <span>
      $49.00
    </span>
  </button>
</DocumentFragment>
`;
